spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -1039,12 +1039,16 @@ protected File get(Message<?> message, Session<F> session, String remoteDir, Str
 			if (!appending && !tempFile.renameTo(localFile)) {
 				throw new MessagingException("Failed to rename local file");
 			}
-			if (this.options.contains(Option.PRESERVE_TIMESTAMP)) {
+			if (this.options.contains(Option.PRESERVE_TIMESTAMP)
+					|| FileExistsMode.REPLACE_IF_MODIFIED.equals(fileExistsMode)) {
 				localFile.setLastModified(getModified(fileInfo));
 			}
 		}
 		else if (FileExistsMode.REPLACE_IF_MODIFIED.equals(fileExistsMode)) {
 			logger.debug("Local file '" + localFile + "' has the same modified timestamp, ignored");
```

Comment:
This DEBUG message should be wrapped with the `if (logger.isDebugEnabled()) {`
OTOH with the new logic via `if (this.command.equals(Command.MGET)) `, I'd say that this message should go to the else side.
WDYT?

I can fix on merge.

I guess since it is a logic change we can't backport it into `4.3.x`